pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -24,6 +24,40 @@ const (
 	PausedAnnotation = "baremetalhost.metal3.io/paused"
 )
 
+// RootDeviceHints holds the hints for specifying the storage location
+// for the root filesystem for the image.
+type RootDeviceHints struct {
+	// A device name like "/dev/vda"
+	DeviceName string `json:"devicename,omitempty"`
+
+	// A SCSI bus address like 0:0:0:0
+	HCTL string `json:"hctl,omitempty"`
+
+	// Device identifier
+	Model string `json:"model,omitempty"`
+
+	// Device vendor
+	Vendor string `json:"vendor,omitempty"`
+
+	// Disk serial number
+	Serial string `json:"serial,omitempty"`
+
+	// Size of the device in Gigabytes
+	SizeGigaBytes int `json:"sizeBytes,omitempty"`
```

Comment:
why not just Size, and document that needs to be gigabyte?

Reply:
These fields end up in the API, and we want users of the API to easily know what sorts of values to fill in. The size for storage is in bytes, but this field uses GB, so being explicit is better.